glommio/src/executor/stall.rs

```diff
@@ -0,0 +1,283 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+//
+#![allow(dead_code)]
+
+use nix::sys;
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread::JoinHandle,
+    time::{Duration, Instant},
+};
+
+#[cfg(feature = "stall-detection")]
+thread_local! {
+    static STALL_STRACE_COLLECTOR: (
+        crossbeam::channel::Sender<backtrace::BacktraceFrame>,
+        crossbeam::channel::Receiver<backtrace::BacktraceFrame>
+    ) = crossbeam::channel::bounded(1 << 10);
+}
+
+#[cfg(feature = "stall-detection")]
+lazy_static! {
+    static ref STALL_SIG_HANDLER: signal_hook::SigId = unsafe {
+        signal_hook::low_level::register(signal_hook::consts::SIGUSR1, move || {
```

Comment:
the documentation should be very explicit that once the stall detector is enabled, the signal is registered.

Reply:
I will make sure the documentation is crystal clear.